plugins/processors/awsapplicationsignals/internal/metrichandlers/aggregation_mutator.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package metrichandlers
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type aggregationType int
+
+const (
+	defaultAggregation aggregationType = iota
+	lastValueAggregation
+)
+
+// AggregationMutator is used to convert predefined ObservableUpDownCounter metrics to use LastValue metrichandlers. This
+// is necessary for cases where metrics are instrumented as cumulative, yet reported with snapshot values.
+//
+// For example, metrics like DotNetGCGen0HeapSize may report values such as 1000, 2000, 1000, with cumulative temporality
+// When exporters, such as the EMF exporter, detect these as cumulative, they convert the values to deltas,
+// resulting in outputs like -, 1000, -1000, which misrepresent the data.
+//
+// Normally, this issue could be resolved by configuring a view with LastValue metrichandlers within the SDK.
+// However, since the view feature is not fully supported in .NET, this workaround implements the required
+// conversion to LastValue metrichandlers to ensure accurate metric reporting.
+// See https://github.com/open-telemetry/opentelemetry-dotnet/issues/2618.
+type AggregationMutator struct {
+	includes map[string]aggregationType
+}
+
+func NewAggregationMutator() AggregationMutator {
+	return newAggregationMutatorWithConfig(map[string]aggregationType{
```

Comment:
Currently this aggregation mutator is specific to DotNet. Do we plan on having this interface be generic to support other languages? 

I would have a config mapping specific to each language

Reply:
Also how does this differ from the current runtime metrics we support? Curious why DotNet requires significant code changes

Reply:
The aggregation is specific to .NET because .NET SDK doesn't provide a mechanism to change the aggregation in the `View` model due to its immaturity. We don't have the same problem in the other languages because their SDKs have provided relatively complete support for the `View` model which allows us to tackle the problem at SDK side.

See https://github.com/open-telemetry/opentelemetry-dotnet/issues/2618.

The aggregation mutator is served as an interim solution before .NET finishes the feature development.

Reply:
See https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2861#issuecomment-2386734618 for how we resolved the same issue in Python SDK.